File: api/src/main/java/com/ruby/rubia_server/core/dto/AILogDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.AILogStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AILogDTO {

    private UUID id;
    private UUID companyId;
    private String companyName;
    private UUID aiAgentId;
    private String aiAgentName;
    private UUID userId;
    private String userName;
    private UUID conversationId;
    private UUID messageId;
    private UUID messageTemplateId;
    private String messageTemplateName;
    private String requestPrompt;
    private String rawResponse;
    private String processedResponse;
    private Integer tokensUsedInput;
    private Integer tokensUsedOutput;
    private BigDecimal estimatedCost;
    private AILogStatus status;
    private String errorMessage;
    private LocalDateTime createdAt;
}

File: api-dotnet/Rubia.Server/DTOs/AILogDto.cs
using Rubia.Server.Enums;
using System.ComponentModel.DataAnnotations;

namespace Rubia.Server.DTOs;

public class AILogDto
{
    public Guid Id { get; set; }
    public Guid CompanyId { get; set; }
    public Guid AiAgentId { get; set; }
    public Guid? UserId { get; set; }
    public Guid? ConversationId { get; set; }
    public Guid? MessageId { get; set; }
    public Guid? MessageTemplateId { get; set; }
    public string RequestPrompt { get; set; } = string.Empty;
    public string? RawResponse { get; set; }
    public string? ProcessedResponse { get; set; }
    public int? TokensUsedInput { get; set; }
    public int? TokensUsedOutput { get; set; }
    public decimal? EstimatedCost { get; set; }
    public AILogStatus Status { get; set; }
    public string? ErrorMessage { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class CreateAILogDto
{
    [Required(ErrorMessage = "Company ID is required")]
    public Guid CompanyId { get; set; }

    [Required(ErrorMessage = "AI Agent ID is required")]
    public Guid AiAgentId { get; set; }

    public Guid? UserId { get; set; }
    public Guid? ConversationId { get; set; }
    public Guid? MessageId { get; set; }
    public Guid? MessageTemplateId { get; set; }

    [Required(ErrorMessage = "Request prompt is required")]
    [MaxLength(10000, ErrorMessage = "Request prompt must not exceed 10000 characters")]
    public string RequestPrompt { get; set; } = string.Empty;

    [MaxLength(10000, ErrorMessage = "Raw response must not exceed 10000 characters")]
    public string? RawResponse { get; set; }

    [MaxLength(10000, ErrorMessage = "Processed response must not exceed 10000 characters")]
    public string? ProcessedResponse { get; set; }

    public int? TokensUsedInput { get; set; }
    public int? TokensUsedOutput { get; set; }
    public decimal? EstimatedCost { get; set; }

    [Required(ErrorMessage = "Status is required")]
    public AILogStatus Status { get; set; }

    [MaxLength(1000, ErrorMessage = "Error message must not exceed 1000 characters")]
    public string? ErrorMessage { get; set; }
}

public class UpdateAILogDto
{
    [MaxLength(10000, ErrorMessage = "Raw response must not exceed 10000 characters")]
    public string? RawResponse { get; set; }

    [MaxLength(10000, ErrorMessage = "Processed response must not exceed 10000 characters")]
    public string? ProcessedResponse { get; set; }

    public int? TokensUsedInput { get; set; }
    public int? TokensUsedOutput { get; set; }
    public decimal? EstimatedCost { get; set; }
    public AILogStatus? Status { get; set; }

    [MaxLength(1000, ErrorMessage = "Error message must not exceed 1000 characters")]
    public string? ErrorMessage { get; set; }
}